test/integration/targets/docker_swarm_facts/tasks/test_swarm_facts.yml

```
@@ -0,0 +1,139 @@
+---
+- block:
+  - name: Make sure we're not already using Docker swarm
+    docker_swarm:
+      state: absent
+      force: true
+
+  - name: Try to get docker_swarm_facts when docker is not running in swarm mode
+    docker_swarm_facts:
+    ignore_errors: yes
+    register: output
+
+  - name: assert failure when called when swarm is not in use or not run on mamager node
+    assert:
+      that:
+         - 'output.failed'
+         - 'output.msg == "Error running docker swarm module: must run on swarm manager node"'
+         - 'output.can_talk_to_docker'
+         - 'output.docker_swarm_active == false'
+         - 'output.docker_swarm_manager == false'
+
+  - name: Create a Swarm cluster
+    docker_swarm:
+      state: present
+    register: output
+
+  - name: assert changed when create a new swarm cluster
+    assert:
+      that:
+         - 'output.changed'
+         - 'output.actions[0] | regex_search("New Swarm cluster created: ")'
+         - 'output.swarm_facts.JoinTokens.Manager'
+         - 'output.swarm_facts.JoinTokens.Worker'
+
+  - name: Try to get docker_swarm_facts when docker is running in swarm mode and as manager
+    docker_swarm_facts:
+    ignore_errors: yes
+    register: output
+
+  - name: assert creding docker swarm facts
+    assert:
+      that:
+         - 'output.docker_swarm_facts.JoinTokens.Manager'
+         - 'output.docker_swarm_facts.JoinTokens.Worker'
+         - 'output.docker_swarm_facts.ID'
+         - 'output.can_talk_to_docker'
+         - 'output.docker_swarm_active'
+         - 'output.docker_swarm_manager'
+
+  - name: Try to get docker_swarm_facts and list of nodes when docker is running in swarm mode and as manager
+    docker_swarm_facts:
+      nodes: yes
+    ignore_errors: yes
+    register: output
+
+  - name: assert reding swarm facts with list of nodes option
+    assert:
+      that:
+         - 'output.docker_swarm_facts.JoinTokens.Manager'
+         - 'output.docker_swarm_facts.JoinTokens.Worker'
+         - 'output.docker_swarm_facts.ID'
+         - 'output.docker_nodes_list'
+         - 'output.docker_nodes_list[0].ID'
```

You probably want
```suggestion
         - 'output.docker_nodes_list[0].ID is string'
```
or something more concrete than a truish value.

---

I think that checking if the node structure has been recorded is sufficient but I don't mind adding this